ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -796,7 +804,9 @@ private void cleanupExpiredInFlightRequests() {
         Map.Entry<Integer, ChunkPutRequestInfo> entry = inFlightRequestsIterator.next();
         if (time.milliseconds() - entry.getValue().startTimeMs > routerConfig.routerRequestTimeoutMs) {
           onErrorResponse(entry.getValue().replicaId);
-          chunkException = new RouterException("Timed out waiting for responses", RouterErrorCode.OperationTimedOut);
+          responseHandler.onRequestResponseException(entry.getValue().replicaId,
+              new IOException("Timed out waiting for a response"));
```

Comment:
Could we just pass this exception into the `RouterException` constructor in cases like this?